packages/flutter/test/material/text_field_test.dart

```diff
@@ -766,6 +771,133 @@ void main() {
     expect(controller.selection.extentOffset, eIndex);
   });
 
+  testWidgets('Read only text field basic', (WidgetTester tester) async {
+    final TextEditingController controller = TextEditingController(text: 'readonly');
+
+    await tester.pumpWidget(
+        overlay(
+          child: TextField(
+            controller: controller,
+            readOnly: true,
+          ),
+        )
+    );
+    // Read only text field cannot open keyboard.
+    await tester.showKeyboard(find.byType(TextField));
+    expect(tester.testTextInput.hasActiveClient, false);
+    await skipPastScrollingAnimation(tester);
+
+    expect(controller.selection.isCollapsed, true);
+
+    await tester.tap(find.byType(TextField));
+    await tester.pump();
+    expect(tester.testTextInput.hasActiveClient, false);
+
+    // Long press on the 'd' character of text 'readOnly'.
+    const int dIndex = 3;
+    final Offset dPos = textOffsetToPosition(tester, dIndex);
+    await tester.longPressAt(dPos);
+    await tester.pump();
```

Comment:
What's the expectation here?  I assume that the readonly still has no text right?  Should we have something like `expect(find.text('d'), findsNothing)`?

Also, why a longPressAt instead of just a tapAt?

Reply:
the text 'readonly' is the initial value, so it does have the text. longpress is required to bring up context menu

Reply:
Ahhhh ok, I get it now.  Maybe update the comment to `// Long press on the 'd' character of 'readonly' to show the context menu`?

Reply:
fixed